llrt_utils/src/encoding.rs

```diff
@@ -71,11 +71,11 @@ impl Encoder {
         }
     }
 
-    pub fn decode(&self, bytes: Vec<u8>) -> Result<Vec<u8>, String> {
+    pub fn decode(&self, bytes: &[u8]) -> Result<Vec<u8>, String> {
```

Comment:
I would use `fn decode<B: Into<Cow<'_, [u8]>>>`(&self, bytes: B`, that will save you a clone on the third branch if you pass in a Vec. AFAIK it is transparent for the caller.